dspace-api/src/main/java/org/dspace/discovery/indexobject/IndexFactoryImpl.java

```diff
@@ -74,32 +79,59 @@ public void writeDocument(Context context, T indexableObject, SolrInputDocument
+                Metadata tikaMetadata = new Metadata();
+                ParseContext tikaContext = new ParseContext();
+
+                // Use Apache Tika to parse the full text stream
+                try {
+                    tikaParser.parse(streams.getStream(), tikaHandler, tikaMetadata, tikaContext);
+                } catch (SAXException saxe) {
+                    // Check if this SAXException is just a notice that this file was longer than the character limit.
+                    // Unfortunately there is not a unique, public exception type to catch here. This error is thrown
+                    // by Tika's WriteOutContentHandler when it encounters a document longer than the char limit
+                    // https://github.com/apache/tika/blob/main/tika-core/src/main/java/org/apache/tika/sax/WriteOutContentHandler.java
+                    if (saxe.getMessage().contains("limit has been reached")) {
+                        // log that we only indexed up to that configured limit
+                        log.info("Full text is larger than the configured limit (discovery.solr.fulltext.charLimit)."
+                                     + " Only the first " + charLimit + " characters were indexed.");
+                    } else {
+                        throw new IOException("Tika parsing error. Could not index full text.", saxe);
+                    }
+                } catch (TikaException ex) {
+                    throw new IOException("Tika parsing error. Could not index full text.", ex);
+                }
 
-                //req.setParam(ExtractingParams.EXTRACT_ONLY, "true");
-                for (String name : doc.getFieldNames()) {
-                    for (Object val : doc.getFieldValues(name)) {
-                        params.add(ExtractingParams.LITERALS_PREFIX + name, val.toString());
+                // Write Tika metadata to "tika_meta_*" fields.
+                // This metadata is not very useful right now, but we'll keep it just in case it becomes more useful.
+                for (String name : tikaMetadata.names()) {
+                    for (String value : tikaMetadata.getValues(name)) {
+                        doc.addField("tika_meta_" + name, value);
                     }
                 }
 
-                req.setParams(params);
-                req.setParam(ExtractingParams.UNKNOWN_FIELD_PREFIX, "attr_");
-                req.setParam(ExtractingParams.MAP_PREFIX + "content", "fulltext");
-                req.setParam(ExtractingParams.EXTRACT_FORMAT, "text");
-                req.setAction(AbstractUpdateRequest.ACTION.COMMIT, true, true);
-                req.process(solr);
+                // Save (parsed) full text to "fulltext" field
+                doc.addField("fulltext", tikaHandler.toString());
+
+                // Add document & commit immediately
+                solr.add(doc);
+                solr.commit(true, true);
```

Comment:
Do we really want a hard commit after every document?  The usual advice from the Solr maintainers on committing in the client is "don't."  If this method is ever used in bulk indexing, the bulk operation will be quite slow *and* make the top-level caches ineffective for all other users.  We should tune our autocommit settings for a good balance of indexing performance and visibility, and trust autocommit.

Reply:
Good point.  To be honest, I debated on this one myself...I only added this in because the code it replaced was also doing a hard commit.  So, I did an exact replacement here.  That said, I'm OK with removing this in favor of autocommit.